### PR TITLE
ET-2340: support card BIN whitelist

### PR DESCRIPTION
**Pull request by jaroslaw-gorecki**

### CODE REVIEW

File: packages/demand-bloc-payment/src/providers/BraintreeProvider.ts [hunk 1]
```diff
@@ -132,6 +150,14 @@ export class BraintreeProvider implements Provider {
     }
   }
 
+  private hasCardBinAllowlist(): boolean {
```
Comment (k4wo) [inline packages/demand-bloc-payment/src/providers/BraintreeProvider.ts:153]: `Allowlist` :D You don't need to be political correct :D

File: packages/demand-bloc-payment/src/providers/BraintreeProvider.ts [hunk 2]
```diff
@@ -132,6 +150,14 @@ export class BraintreeProvider implements Provider {
...
+    return Array.isArray(this.options.allowedBinValues) && this.options.allowedBinValues.length > 0
+  }
+
+  private isCardBinAllowed(bin: string | undefined): boolean {
+    return this.hasCardBinAllowlist() ? !!bin && !!this.options.allowedBinValues?.includes(bin) : true
```
Comment (k4wo) [inline packages/demand-bloc-payment/src/providers/BraintreeProvider.ts:158]: `!!` is redundant here `!!this.options.allowedBinValues?.includes(bin)`

**Review by k4wo — APPROVED** [review]

**Review by wkoszyk-karhoo — APPROVED** [review]

File: packages/demand-bloc-payment/src/providers/BraintreeProvider.ts [hunk 3]
```diff
@@ -80,6 +82,7 @@ export class BraintreeProvider implements Provider {
       withThreeDSecure,
     } = this.options
 
+    this.binValue = undefined
```
[the remark this answers is not in the document]
Comment (jaroslaw-gorecki) [reply packages/demand-bloc-payment/src/providers/BraintreeProvider.ts:85]: we need - every time you call initialize it should reset

File: packages/demand-bloc-payment/src/providers/BraintreeProvider.ts (hunk 2 above)
Comment (jaroslaw-gorecki) [reply packages/demand-bloc-payment/src/providers/BraintreeProvider.ts:158]: TS issue without `!!` - expects boolean, not `boolean | undefined`

**Pull request merged**